src/core/runtime/runtime.cc

```diff
@@ -76,6 +80,10 @@ static const char* const core_library_name = "legate.core";
 
   const char* sync_stream_view = getenv("LEGATE_SYNC_STREAM_VIEW");
   if (sync_stream_view != nullptr && atoi(sync_stream_view) > 0) synchronize_stream_view = true;
+
+#ifdef LEGATE_USE_CUDA
+  nvtx::Range::initialize(core_library_name);
```

Comment:
I think we would want a separate `Domain` per Legate library, instead of putting all ranges under the same domain. A prerequisite for that is a centralized registry of all legate libraries, which doesn't exist in this branch. we can certainly merge this PR and tackle the issue in a follow-on PR, as the registry can take some time to implement properly.

Reply:
After consulting with @jcohen-nvidia, we think your idea would be better suited to the Categories feature in NVTX rather than creating separate domains, which would come with a performance cost and also complicate the implementation of a toggle switch for tracing Legate in Nsight Systems.

Reply:
I repeat here what I said in the meeting to remind ourselves: I still think each of Legate libraries should have a separate `Domain` , as they are physically different libraries with separate APIs that the user may or may not choose to trace, and each library might want to use its own set of categories for the domain operations. We can merge this PR as-is for now and I can make a follow-on PR for a per-library `Domain` (which would naturally fit to this `LibraryContext`: https://github.com/nv-legate/legate.core/blob/branch-22.07/src/core/runtime/context.h#L65-L105).